### PR TITLE
Depth

### PR DESCRIPTION
**Pull request by AnnaLangner**
Add depth parameter to find links command

### CODE REVIEW

File: script/main.py [hunk 1]
```diff
@@ -1,46 +1,45 @@
...
 
 def main():
-    driver, url = fetch_argument()
+    driver, url, depth = fetch_argument()
     driver = create_driver(driver)
-    driver.get(url)
-    WebDriverWait(driver, 10).until(EC.visibility_of_all_elements_located(By.TAG_NAME, "a"))
-    links = find_links(driver)
-    print_link_info(links)
+    urls = [url]
```
Comment (klangner) [inline script/main.py]: It is better to use set instead of list, since on typical page you can expect duplicate links

File: script/main.py [hunk 2]
```diff
@@ -1,46 +1,45 @@
...
 
 
 def main():
-    driver, url = fetch_argument()
+    driver, url, depth = fetch_argument()
     driver = create_driver(driver)
-    driver.get(url)
-    WebDriverWait(driver, 10).until(EC.visibility_of_all_elements_located(By.TAG_NAME, "a"))
-    links = find_links(driver)
-    print_link_info(links)
```
Comment (klangner) [inline script/main.py:43]: This program should print list of found links. It looks that this was removed, why?

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 2 above)
Comment (AnnaLangner) [reply script/main.py:43]: I thought it was not necessary

File: script/main.py [hunk 3]
```diff
@@ -1,46 +1,52 @@
...
+    for elem in range(depth + 1):
+        new_urls = set()
+        for url in urls:
+            driver.get(url)
+            WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.TAG_NAME, 'a')))
+            links = find_links(driver)
+            for item in links:
+                new_urls.add(item)
+        urls = new_urls
+    print_urls_info(urls)
```
Comment (klangner) [inline script/main.py]: Please notice that you don't print all links. Only from the last page.
You should print all unique links

**Review by klangner — CHANGES_REQUESTED** [review]

**Review by klangner — APPROVED** [review]

**Pull request merged**